### PR TITLE
Diagnostics include column integrated tracer 

### PR DESCRIPTION
**Pull request by elynnwu**
## Purpose

#217 

We can now have column integrated tracers (2D) in model output:
```
diagnostics_config:
  path: "output.zarr"
  names:
    - u
    - v
    - ...
  derived_names:
    - column_integrated_qvapor
    - column_integrated_qliquid
```

## Code changes:

- diagnostics.py: `derived_names` can now be added to specify variables that are derived from dycore or physics state
  - Currently, only `column_integrated_*`  tracers are supported.
- plot_pcolormesh_cube.py: add support for plotting 2D variable by passing `--var2D`
## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by elynnwu** [comment]
launch jenkins

File: driver/pace/driver/diagnostics.py [hunk 1]
```diff
@@ -102,9 +110,26 @@ def store(self, time: Union[datetime, timedelta], state: DriverState):
...
+            for name in self.derived_names:
+                if "column_integrated_" in name:
+                    tracer = name.split("column_integrated_")[-1]
+                    output[name] = _compute_column_integral(
+                        name,
+                        getattr(state.dycore_state, tracer),
+                        state.dycore_state.delp,
+                    )
+                else:
+                    print(f"{name} is not a supported diagnostic variable.")
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:130]: should this be just a print or an error?

File: driver/examples/plot_pcolormesh_cube.py [hunk 2]
```diff
@@ -95,7 +95,11 @@ def parse_args():
         help="ending time step",
         default=1,
     )
-
+    parser.add_argument(
+        "--var2D",
+        action="store_true",
+        help="whether variable is 2D",
```
Comment (FlorianDeconinck) [inline driver/examples/plot_pcolormesh_cube.py:101]: whether variable is 2D, for diagnostics

File: driver/pace/driver/diagnostics.py (hunk 1 above)
Comment (FlorianDeconinck) [inline driver/pace/driver/diagnostics.py:130]: `warn` probably better than print

**Review by FlorianDeconinck — APPROVED** [review]

File: driver/pace/driver/diagnostics.py [hunk 3]
```diff
@@ -102,9 +110,26 @@ def store(self, time: Union[datetime, timedelta], state: DriverState):
...
+            zarr_state.update(derived_state)
             assert time is not None
             self.monitor.store(zarr_state)
 
+    def _get_derived_state(self, state: DriverState):
+        output = {}
+        if len(self.derived_names) > 0:
+            for name in self.derived_names:
+                if "column_integrated_" in name:
+                    tracer = name.split("column_integrated_")[-1]
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:123]:
```suggestion
                    tracer = tracer[len("column_integrated_"):]
```

File: driver/pace/driver/diagnostics.py [hunk 4]
```diff
@@ -102,9 +110,26 @@ def store(self, time: Union[datetime, timedelta], state: DriverState):
...
+            derived_state = self._get_derived_state(state)
+            zarr_state.update(derived_state)
             assert time is not None
             self.monitor.store(zarr_state)
 
+    def _get_derived_state(self, state: DriverState):
+        output = {}
+        if len(self.derived_names) > 0:
+            for name in self.derived_names:
+                if "column_integrated_" in name:
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:122]:
```suggestion
                if name.startswith("column_integrated_"):
```

File: driver/pace/driver/diagnostics.py [hunk 5]
```diff
@@ -131,3 +156,26 @@ def store_grid(
...
+    Compute column integrated mixing ratio (e.g., total liquid water path)
+
+    Args:
+        q_in: tracer mixing ratio
+        delp: pressure thickness of atmospheric layer
+    """
+    column_integral = pace.util.Quantity(
+        sum(
+            q_in.data[:, :, k] * delp.data[:, :, k]
+            for k in range(q_in.metadata.extent[2])
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:174]: do we need to check that the extents are the same?

File: driver/pace/driver/diagnostics.py [hunk 6]
```diff
@@ -131,3 +156,26 @@ def store_grid(
         self, grid_data: pace.util.grid.GridData, metadata: QuantityMetadata
     ):
         pass
+
+
+def _compute_column_integral(
+    name: str, q_in: pace.util.Quantity, delp: pace.util.Quantity
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:162]: should this specifically be q and delp or should these be more generic?

**Review by twicki — CHANGES_REQUESTED** [review]

File: driver/pace/driver/diagnostics.py (hunk 1 above)
Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:130]: Changed to warning

Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:130]: Changed to warning, my thought is that we would still run the model and just not save any unsupported diagnostic variables.

File: driver/pace/driver/diagnostics.py (hunk 3 above)
Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:123]: I think it should be `name[len("column_integrated_"):]`

File: driver/pace/driver/diagnostics.py (hunk 5 above)
Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:174]: This is a sum over the k-axis, so it should error if you try to put a 2D variable.

Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:174]: Will add error message

File: driver/pace/driver/diagnostics.py (hunk 6 above)
Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:162]: This only works for column integrating tracers, for other column integrated variable, other factors are needed. For now, we only support column integrated tracers.

**Comment by elynnwu** [comment]
launch jenkins

**Comment by elynnwu** [comment]
launch jenkins

**Pull request merged**